travel-event-host-app/src/app/api/event/[id]/register/route.ts

```diff
@@ -0,0 +1,27 @@
+import { NextResponse } from 'next/server';
+import { connectMongoDB } from '@/lib/mongodb';
+import Event from '@/schemas/event';
+import mongoose from 'mongoose';
+
+export async function PATCH(req: Request, params: any) {
+  let { userId } = await req.json();
+
+  const id = params.id;
+  await connectMongoDB();
+  const isValidObjectId = mongoose.Types.ObjectId.isValid(id);
+  if (!isValidObjectId) {
+    return NextResponse.json({ message: 'Invalid ObjectId format' }, { status: 400 });
+  }
+
+  const eventFound = await Event.findById(id);
+
+  if (eventFound) {
+    eventFound.participantIds.push({ userId, timeStamp: Date.now() });
```

Comment:
Before we push the userId to this participantIds array, I think we should check if the user's ID isn't already in the array.

Reply:
yeah